### PR TITLE
Status Message Improvements

### PR DESCRIPTION
**Pull request by Bubu**
~~Put the status in parenthesis, which is the usual presentation for a user status i.e. UserX (Away).~~

Also significantly reduce the font size as to visually indicate this is additional information and not part of the username or message text.

Add a hover text to indicate this is a status message.

Screenshots:

| Before         | After     |
|--------------|-----------|
| image | image      |

UserProfile:

image

### CODE REVIEW

**Comment by Bubu** [comment]
Still undecided between 0.75 and 0.8 scaling factor.  :D

**Comment by deepbluev7** [comment]
I would probably leave out the () :)

**Comment by deepbluev7** [comment]
And afaik we use 0.8 elsewhere, but you might want to Math.floor it

**Comment by LorenDB** [comment]
> I find that almost the more important part of this commit tbh. It's how people already encode their status into the nickname currently and is how ICQ did it many years back as well. 🤷 (To clarify: Icq had proper statuses, but when displaying it I remember it always being in parenthesis behind the nickname)

IMO if you are going to modify what the status message shows, it should be limited to prepending "Status: " to the status. But I think that it is fairly obvious that the status message is a status message already, so you really shouldn't have to modify it. Alternately, you could just add a tooltip that says something about this being a status.

**Comment by Bubu** [comment]
I removed the parentheses, added a hover text and added the status message text to the userprofile page.

File: resources/qml/dialogs/UserProfile.qml [hunk 1]
```diff
@@ -56,7 +56,7 @@ ApplicationWindow {
             id: contentL
 
             width: devicelist.width
-            spacing: 10
+            spacing: 8
```
Comment (LorenDB) [inline resources/qml/dialogs/UserProfile.qml:59]: Can you make this one of the paddings specified in NhekoGlobalObject.h?

**Review by LorenDB — CHANGES_REQUESTED** [review]
This looks good overall to me.

File: resources/qml/MessageView.qml [hunk 2]
```diff
@@ -373,12 +374,21 @@ Item {
...
-                        width: userInfo.remainingWidth - userName_.width - parent.spacing
+                        width: Math.min(implicitWidth, userInfo.remainingWidth - userName_.width - parent.spacing)
                         font.italic: true
+                        font.pointSize: Math.floor(fontMetrics.font.pointSize * 0.8)
+                        ToolTip.text: qsTr("%1's status message").arg(userName)
+                        ToolTip.visible: status_label_hover_handler.hovered
+                        ToolTip.delay: Nheko.tooltipDelay
+
+                        HoverHandler {
+                            id: status_label_hover_handler
```
Comment (Bubu) [inline resources/qml/MessageView.qml:390]: heh, python casing slipped in here, I'll change that for consistency :)

**Review by LorenDB — APPROVED** [review]
LGTM

**Comment by deepbluev7** [comment]
> I've seen many examples in the codebase that don't do this. What's the benefit of adding it?

I have seen alignment break horribly when you get some crooked font sizes, that are not a multiple of 0.5. It probably isn't needed, I guess I should investigate more, why I originally added that.

**Comment by deepbluev7** [comment]
grafik

grafik

I think I found a bug

File: resources/qml/dialogs/UserProfile.qml [hunk 3]
```diff
@@ -177,6 +177,21 @@ ApplicationWindow {
                 Layout.alignment: Qt.AlignHCenter
             }
 
+            MatrixText {
+                id: statusMsg
+                text: qsTr("Status: %1").arg(Presence.userStatus(profile.userid))
+                visible: Presence.userStatus(profile.userid) != ""
+                Layout.alignment: Qt.AlignHCenter
```
Comment (deepbluev7) [inline resources/qml/dialogs/UserProfile.qml:184]:
```suggestion
                Layout.alignment: Qt.AlignHCenter
                Layout.fillWidth: true
                horizontalAlignment: TextEdit.AlignHCenter
```

**Comment by deepbluev7** [comment]
Otherwise it looks good to me :3

**Comment by deepbluev7** [comment]
Thank you!

**Pull request merged**